### PR TITLE
STG-11: Event Details Screen

### PR DESCRIPTION
**Pull request by ToniVinter**

### CODE REVIEW

File: lib/app/features/event/application/event_notifier.dart [hunk 1]
```diff
@@ -22,11 +22,33 @@ class EventNotifier extends _$EventNotifier {
...
     }
     logger.i('init event provider state');
+    await getPlaylist();
     await getPastEvents();
     await getThisWeekEvents();
     await getUpcomingEvents();
   }
 
+  Future<void> getEventById(String eventId) async{
+   // ref.read(loadingProvider.notifier).state = true;
```
Comment (ToniVinter) [inline lib/app/features/event/application/event_notifier.dart:32]: You can uncomment these

File: lib/app/features/event/data/event_repository.dart [hunk 2]
```diff
@@ -50,6 +60,34 @@ class EventRepository extends _$EventRepository {
     return [];
   }
 
+  Future<EventModel?> getEventById(String eventId) async {
+    try {
+      final uri = API.getEvent(eventId); // Replace with your API endpoint for fetching a single event by ID
```
Comment (ToniVinter) [inline lib/app/features/event/data/event_repository.dart:65]: Delete commented code

File: lib/app/features/event/presentation/event_details_screen.dart [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+          },
+        ),
+        title: Text(_event?.name ?? ''),
+      ),
+       body: isLoading ? _buildLoadingIndicator() :_buildContent(context),
+    );
+  }
+
+  Widget _buildContent(BuildContext context) {
+    _playlist = ref.watch(eventNotifierProvider).playlist;
```
Comment (ToniVinter) [inline lib/app/features/event/presentation/event_details_screen.dart:60]: move these fields to build() ( near isLoading and _event)

File: lib/app/features/event/presentation/event_details_screen.dart [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      padding: defaultScreenPadding,
+      child: ListView(
+        children: [
+          const SizedBox(height: Insets.medium),
+          Text(
+            formattedDate,
+            style: context.textTheme.bodyMedium,
+          ),
+          const SizedBox(height: Insets.medium),
+          SingleChildScrollView(
```
Comment (ToniVinter) [inline lib/app/features/event/presentation/event_details_screen.dart:73]: That's not the best approach to use for a horizontal listing

You have to use : Listview.builder() 

And build the widget from a list of Participants ( from dummy or something )

File: lib/app/features/event/presentation/event_details_screen.dart [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+          if (ref.watch(loadingProvider.notifier).state)
+            _buildLoadingIndicator()
+          else
+            _buildPlaylist(),
+        ],
+      ),
+    );
+  }
+
+  ListView _buildPlaylist() {
```
Comment (ToniVinter) [inline lib/app/features/event/presentation/event_details_screen.dart:108]: a best approach would be that every _buildETC() method to have as returnig type Widget, will look like this:|
Widget _buildPlaylist()

File: lib/app/features/event/presentation/event_details_screen.dart [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+          children: [
+            SongChordTile(song: song),
+            const SizedBox(height: Insets.smallNormal),
+          ],
+        );
+      },
+    );
+  }
+
+  Center _buildLoadingIndicator() {
```
Comment (ToniVinter) [inline lib/app/features/event/presentation/event_details_screen.dart:126]: same here, use Widget

File: lib/app/features/event/presentation/event_details_screen.dart [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        width: 50,
+        height: 50,
+        child: LoadingIndicator(
+          indicatorType: Indicator.ballClipRotateMultiple,
+        ),
+      ),
+    );
+  }
+
+  Widget _buildParticipants() {
```
Comment (ToniVinter) [inline lib/app/features/event/presentation/event_details_screen.dart:138]: I think this is not used, but this is the approach you have to use for participants

File: lib/app/features/event/presentation/events_screen.dart [hunk 8]
```diff
@@ -142,9 +142,18 @@ class EventsScreenState extends ConsumerState<EventsScreen> {
 
         return Column(
           children: [
-            EventTile(
-              title: event.name,
-              description: formattedDate,
+            GestureDetector(
+              onTap: () {
+                // Navigator.pushNamed(context, AppRoute.eventDetails.name);
```
Comment (ToniVinter) [inline lib/app/features/event/presentation/events_screen.dart:147]: commented code, we need this?

File: lib/app/shared/participant_profile.dart [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  final String fullName;
+
+  ParticipantProfile({required this.profilePicture, required this.fullName});
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Container(
+          width: 48, // You can adjust the width as needed
```
Comment (ToniVinter) [inline lib/app/shared/participant_profile.dart:15]: delete commented code

File: lib/app/shared/participant_profile.dart [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+          height: 48, // This sets the circular avatar height
+          decoration: BoxDecoration(
+            shape: BoxShape.circle,
+            image: DecorationImage(
+              fit: BoxFit.cover,
+              image: AssetImage(profilePicture), // Use AssetImage for local assets
```
Comment (ToniVinter) [inline lib/app/shared/participant_profile.dart:21]: commented code

File: lib/app/shared/participant_profile.dart [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+            ),
+          ),
+        ),
+        SizedBox(height: 8), // Adjust the spacing between the circle and text
```
Comment (ToniVinter) [inline lib/app/shared/participant_profile.dart:25]: commented code

File: lib/app/shared/song_chord_tile.dart [hunk 12]
```diff
@@ -1,25 +1,31 @@
...
 import 'package:on_stage_app/app/utils/build_context_extensions.dart';
 
-class SongChordTile extends StatelessWidget {
+class SongChordTile extends StatefulWidget {
   const SongChordTile({
-    required this.title,
-    required this.description,
-    required this.chord,
+    required this.song,
+    // required this.title,
```
Comment (ToniVinter) [inline lib/app/shared/song_chord_tile.dart:10]: commented code

File: lib/app/shared/song_chord_tile.dart [hunk 13]
```diff
@@ -1,25 +1,31 @@
...
-  final String description;
-  final String chord;
+  final SongModel song;
 
+  @override
+  State<SongChordTile> createState() => _SongChordTileState();
+}
+
+class _SongChordTileState extends State<SongChordTile> {
+  // final String title;
```
Comment (ToniVinter) [inline lib/app/shared/song_chord_tile.dart:23]: commented code

File: lib/app/features/event/presentation/event_details_screen.dart [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+    final isLoading = ref.watch(eventNotifierProvider).event == null;
+
+    _event = ref.watch(eventNotifierProvider).event;
+    return  Scaffold(
+      appBar: AppBar(
+        backgroundColor: context.colorScheme.background,
+        leading: IconButton(
+          icon: Icon(Icons.arrow_back),
+          onPressed: () {
+            Navigator.of(context).pop();
```
Comment (ToniVinter) [inline lib/app/features/event/presentation/event_details_screen.dart:50]: use context.pop(), we only use Navigator for modals/ popups

File: lib/app/features/event/application/event_notifier.dart [hunk 15]
```diff
@@ -22,11 +22,33 @@ class EventNotifier extends _$EventNotifier {
...
+    final event = await ref
+    .read(eventRepositoryProvider.notifier)
+    .getEventById(eventId);
+
+    state = state.copyWith(event: event);
+    //ref.read(loadingProvider.notifier).state = false;
```
Comment (ToniVinter) [inline lib/app/features/event/application/event_notifier.dart:38]: commented code

**Review by ToniVinter — COMMENTED** [review]
@EmFarcau , sa te uiti peste comments

File: lib/app/features/event/presentation/event_details_screen.dart (hunk 7 above)
Comment (EmFarcau) [reply lib/app/features/event/presentation/event_details_screen.dart:138]: This is how i wanted in the beginning 🍡

**Pull request merged**